ui/ui-components/components/record/SampleRecordCard.tsx

```diff
@@ -46,30 +48,64 @@ export interface SampleRecordCardProps {
   reloadKey?: string;
   warning?: string;
   groupId?: string;
+
+  record?: RecordType;
+  groups?: Models.GroupType[];
+  destinations?: Models.DestinationType[];
 }
 
 // This is exported so we can manipulate it in tests, but is unused elsewhere.
 export const isConfigUI = grouparooUiEdition() === "config";
 
+const cookies = new Cookies();
+
 const getCachedSampleRecordId = (modelId: string): string => {
-  return globalThis.localStorage?.getItem(
-    `sampleRecord:${grouparooUiEdition()}:${modelId}`
-  );
+  if (isBrowser()) {
+    const localStorageRecordId = globalThis.localStorage?.getItem(
+      `sampleRecord:${grouparooUiEdition()}:${modelId}`
+    );
+    if (localStorageRecordId) {
+      setCachedSampleRecordId(modelId, localStorageRecordId);
+      globalThis.localStorage?.removeItem(
+        `sampleRecord:${grouparooUiEdition()}:${modelId}`
+      );
+      return localStorageRecordId;
+    } else {
+      return cookies.get(`sampleRecord:${grouparooUiEdition()}`)?.[modelId];
+    }
+  }
+  return undefined;
 };
 
 const setCachedSampleRecordId = (modelId: string, recordId: string): void => {
-  if (recordId) {
-    globalThis.localStorage?.setItem(
-      `sampleRecord:${grouparooUiEdition()}:${modelId}`,
-      recordId
+  if (isBrowser() && recordId) {
+    const existing = cookies.get<Record<string, string>>(
+      `sampleRecord:${grouparooUiEdition()}`
+    );
+    cookies.set(
```

Comment:
Does this mean we should switch to only using cookies and stop using local storage?

Reply:
In this case we've switched to cookies so that both client-side and server-side rendering has access to the user's cached sample record. 

The logic here is to "migrate" existing client side state into the cookie when the user first navigates to the page.

Reply:
The only other explicit usage of localstorage in our codebase is the `session:csrfToken`, which seems fine.